simpeg_drivers/utils/tile_estimate.py

```diff
@@ -0,0 +1,186 @@
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#  All rights reserved.
+#
+#  This file is part of simpeg-drivers.
+#
+#  The software and information contained herein are proprietary to, and
+#  comprise valuable trade secrets of, Mira Geoscience, which
+#  intend to preserve as trade secrets such software and information.
+#  This software is furnished pursuant to a written license agreement and
+#  may be used, copied, transmitted, and stored only in accordance with
+#  the terms of such license and with the inclusion of the above copyright
+#  notice.  This software and information or any other copies thereof may
+#  not be provided or otherwise made available to any other person.
+#
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+import sys
+from pathlib import Path
+from typing import ClassVar
+
+import matplotlib.pyplot as plt
+import numpy as np
+from geoapps_utils.driver.data import BaseData
+from geoapps_utils.utils.numerical import fibonacci_series, fit_circle
+from geoh5py.groups import SimPEGGroup, UIJsonGroup
+from geoh5py.ui_json import InputFile
+from geoh5py.ui_json.utils import fetch_active_workspace, monitored_directory_copy
+from scipy.interpolate import interp1d
+from tqdm import tqdm
+
+from simpeg_drivers import assets_path
+from simpeg_drivers.driver import DRIVER_MAP, InversionDriver
+from simpeg_drivers.utils.utils import (
+    active_from_xyz,
+    simpeg_group_to_driver,
+    tile_locations,
+)
+
+
+class TileEstimator:
+    def __init__(self, input_file: str | Path | InputFile):
+        if isinstance(input_file, str | Path):
+            input_file = InputFile.read_ui_json(input_file)
+
+        if not isinstance(input_file, InputFile):
+            raise ValueError(
+                "Input must be a path to a UI JSON file or an InputFile instance."
+            )
+
+        self.params = TileParameters.build(input_file)
+
+    def start(self) -> SimPEGGroup:
+        """
+        Run the tile estimator.
+        """
+        driver = simpeg_group_to_driver(self.params.simulation, self.params.geoh5)
+        mesh = driver.inversion_mesh.mesh
+        locations = driver.inversion_data.locations
+        active_cells = active_from_xyz(
+            driver.inversion_mesh.entity, locations, method="nearest"
+        )
+        results = {}
+        counts = fibonacci_series(15)[2:]
+        for count in tqdm(counts, desc="Estimating tiles:"):
+            tiles = tile_locations(
```

Comment:
Can we break this into smaller pieces?  

Something like:

``` 
for count in tqdm():
    results[count] = self._get_result()

new_out_group = get_new_out_group(results)
return new_out_group
```
with private helper methods for lines 71 and 83 just to explain what they do in the docstring/typing and another helper for the `if params.render_plot:` block.

Reply:
..and unit tests please!